definitions/features/puppet_server.rb

```diff
@@ -2,19 +2,58 @@ class Features::PuppetServer < ForemanMaintain::Feature
   metadata do
     label :puppet_server
 
-    # We only check puppetserver and not puppet-server, as puppet-server
-    # is a part of httpd and relies on httpd service to restart, therefore
-    # not requiring a separate service to restart
     confine do
-      find_package('puppetserver') || find_package('puppet')
+      find_package('puppet-server') || find_package('puppetserver') || find_package('puppet')
     end
   end
 
+  def config_files
+    [
+      '/etc/puppet',
+      '/etc/puppetlabs',
+      '/opt/puppetlabs/puppet/cache/foreman_cache_data',
+      '/var/lib/puppet/foreman_cache_data',
+      '/opt/puppetlabs/puppet/ssl/',
+      '/var/lib/puppet/ssl',
+      '/var/lib/puppet'
+    ]
+  end
+
   def services
+    # We only check puppetserver and not puppet-server, as puppet-server
+    # is a part of httpd and relies on httpd service to restart, therefore
+    # not requiring a separate service to restart
     find_package('puppetserver') ? { 'puppetserver' => 30 } : {}
   end
 
   def puppet_version
     version(execute!('puppet --version'))
   end
+
+  def find_empty_cacert_request_files
+    cmd_output = execute!("find #{cacert_requests_directory} -type f -size 0 | paste -d, -s")
+    cmd_output.split(',')
+  end
+
+  def delete_empty_cacert_files
+    execute!("find #{cacert_requests_directory} -type f -size 0 -delete")
+  end
+
+  def cacert_requests_directory
+    "#{ca_directory_path}/requests"
+  end
+
+  def cacert_requests_dir_exists?
+    File.directory?(cacert_requests_directory)
+  end
+
+  private
+
+  def ca_directory_path
+    "#{puppet_ssldir_path}/ca"
+  end
+
+  def puppet_ssldir_path
+    execute!('puppet config print ssldir')
```

Comment:
This execute throws warning and redirect 2> /dev/null is not working here, request to check this one.

Reply:
@kgaikwad, @upadhyeammit, @iNecas I found out we are not the only ones suffering from this [issue](https://tickets.puppetlabs.com/browse/PUP-8566) and we can as an alternative use 
```
 puppet master --configprint ssldir
``` 
I positively tested it with puppet 4,5 but have no version 3. It would be great if someone could confirm.